packages/flutter/test/material/popup_menu_theme_test.dart

```diff
@@ -0,0 +1,295 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('PopupMenuThemeData copyWith, ==, hashCode basics', () {
+    expect(const PopupMenuThemeData(), const PopupMenuThemeData().copyWith());
+    expect(const PopupMenuThemeData().hashCode, const PopupMenuThemeData().copyWith().hashCode);
+  });
+
+  test('PopupMenuThemeData null fields by default', () {
+    const PopupMenuThemeData popupMenuTheme = PopupMenuThemeData();
+    expect(popupMenuTheme.color, null);
+    expect(popupMenuTheme.shape, null);
+    expect(popupMenuTheme.elevation, null);
+    expect(popupMenuTheme.textStyle, null);
+  });
+
+  testWidgets('Default PopupMenuThemeData debugFillProperties',
+          (WidgetTester tester) async {
+    final DiagnosticPropertiesBuilder builder = DiagnosticPropertiesBuilder();
+    const PopupMenuThemeData().debugFillProperties(builder);
+
+    final List<String> description = builder.properties
+        .where((DiagnosticsNode node) => !node.isFiltered(DiagnosticLevel.info))
+        .map((DiagnosticsNode node) => node.toString())
+        .toList();
+
+    expect(description, <String>[]);
+  });
+
+  testWidgets('PopupMenuThemeData implements debugFillProperties',
+          (WidgetTester tester) async {
+    final DiagnosticPropertiesBuilder builder = DiagnosticPropertiesBuilder();
+    PopupMenuThemeData(
+      color: const Color(0xFFFFFFFF),
+      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(2.0)),
+      elevation: 2.0,
+      textStyle: const TextStyle(color: Color(0xffffffff)),
+    ).debugFillProperties(builder);
+
+    final List<String> description = builder.properties
+        .where((DiagnosticsNode node) => !node.isFiltered(DiagnosticLevel.info))
+        .map((DiagnosticsNode node) => node.toString())
+        .toList();
+
+    expect(description, <String>[
+      'color: Color(0xffffffff)',
+      'shape: RoundedRectangleBorder(BorderSide(Color(0xff000000), 0.0, BorderStyle.none), BorderRadius.circular(2.0))',
+      'elevation: 2.0',
+      'text style: TextStyle(inherit: true, color: Color(0xffffffff))'
+    ]);
+  });
+
+  testWidgets('Passing no PopupMenuThemeData returns defaults',
+          (WidgetTester tester) async {
+    final Key popupButtonKey = UniqueKey();
+    final Key popupButtonApp = UniqueKey();
+    final Key popupItemKey = UniqueKey();
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(),
+      key: popupButtonApp,
+      home: Material(
+        child: Column(
+          children: <Widget>[
+            PopupMenuButton<void>(
```

Comment:
I don't think it's necessary to specify `<void>` here if the analyzer allows you to remove it

Reply:
The analyzer says that we can't remove `<void>` because [type annotations must be specified](https://dart-lang.github.io/linter/lints/always_specify_types.html).